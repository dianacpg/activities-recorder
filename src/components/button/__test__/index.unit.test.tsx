import { expect, describe, it, vi } from "vitest";
import { render, fireEvent } from "@testing-library/react";
// Components
import Button, { ButtonProps } from "..";

describe("Button component", () => {
  const testComponent = (skin?: ButtonProps["skin"], onClick?: ButtonProps["onClick"]) =>
    render(
      <Button skin={skin} onClick={onClick}>
        Click Me
      </Button>
    );
  it("renders the Button component", () => {
    expect(testComponent).toMatchSnapshot();
  });

  it("applies the 'ghost' skin", () => {
    expect(testComponent("ghost")).toMatchSnapshot();
  });

  it("fires click event when button is clicked", () => {
    const onClickMock = vi.fn();
    const { getByText } = testComponent("ghost", onClickMock);
    const buttonElement = getByText("Click Me");
    fireEvent.click(buttonElement);
    expect(onClickMock).toHaveBeenCalled();
  });
});
