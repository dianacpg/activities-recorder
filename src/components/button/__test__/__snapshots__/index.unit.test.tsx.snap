// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > applies the 'ghost' skin 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="button--ghost"
      >
        Click Me
      </button>
    </div>
    <div>
      <button
        class="button--ghost"
      >
        Click Me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="button--ghost"
    >
      Click Me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component > renders the Button component 1`] = `[Function]`;
